src/adapters/error.ts

```diff
@@ -0,0 +1,109 @@
+import { z, ZodError } from "zod";
+import * as platform from "platform";
+import * as StackTrace from "stacktrace-js";
+
+import { StrictSchema } from "src/utils/type-safety";
+
+interface MessageKeyError {
```

Comment:
Should we add this interfaces to our domain?

Reply:
We have discussed this topic in the past for other types. This is the definition of the Domain that I wrote [some time ago](https://www.notion.so/iden3/Wallet-UI-Typescript-refactor-f8bd6c2fe6db46ffb40ff2419441fed6#5c35d68155cd43d1b42d5b1d574247d6):

> The domain layer contains the abstract definition of the types that are used across the application. These types should generally correspond to real entities of the domain of the application. Special care must be taken on the decision of what constitutes a domain entity, as these types will determine the general discourse of the code of the whole application. Think of it as the glossary of the application.

So no, I don't think we should add `MessageKeyError` to the domain since it's not a central type of the application. It's just an accessory type used by an error parser in `adapters`.